air-quality-ui/e2e/vAirify_summary_page.ts

```diff
@@ -0,0 +1,162 @@
+import { type Locator, type Page, expect } from '@playwright/test'
+
+export class VairifySummaryPage {
+  readonly page: Page
+  readonly apiForecast: object
+  readonly apiSummary: object
+  readonly title: Locator
+  readonly scroller: Locator
+  readonly agCell: Locator
+  readonly allCells: Locator
+  readonly kampalaBtn: Locator
+  readonly abuDhabiBtn: Locator
+  readonly zurichBtn: Locator
+  readonly citiesBtn: Locator
+
+  constructor(page: Page, apiForecast: object, apiSummary: object) {
+    this.page = page
+    this.apiForecast = apiForecast
+    this.apiSummary = apiSummary
+    this.title = this.page.locator('title')
+    this.scroller = this.page.locator('.ag-body-horizontal-scroll-viewport')
+    this.agCell = this.page.locator('role=gridcell')
+    this.allCells = this.page.locator('[role=gridcell]')
+    this.kampalaBtn = this.page.getByRole('link', { name: 'Kampala' })
+    this.abuDhabiBtn = this.page.getByRole('link', { name: 'Abu Dhabi' })
+    this.zurichBtn = this.page.getByRole('link', { name: 'Zurich' })
+    this.citiesBtn = this.page.getByRole('link', { name: 'Cities' })
+  }
+  async clickButton(buttonName: string) {
+    switch (buttonName) {
```

Comment:
Maybe I'm missing something but is this going to be painful as we get more buttons, could we replace all of this by:
async clickButton(buttonName: string) {
  await this.page.getByRole('link', {name: buttonName})
}

?

Reply:
This is my fault - yes that's much nicer